WooCommerce/Classes/ViewModels/Order Details/OrderDetailsViewModel.swift

```diff
@@ -601,7 +607,33 @@ extension OrderDetailsViewModel {
         stores.dispatch(action)
     }
 
-    func refreshCardPresentPaymentEligibility() {
+    func checkCardPresentPaymentEligibility(onCompletion: @escaping (() -> Void)) {
+        let configuration = configurationLoader.configuration
+
+        guard configuration.isSupportedCountry else {
+            dataSource.isEligibleForCardPresentPayment = false
+            onCompletion()
+            return
+        }
+
+        let action = OrderCardPresentPaymentEligibilityAction
+            .orderIsEligibleForCardPresentPayment(orderID: order.orderID,
+                                                  siteID: order.siteID,
+                                                  cardPresentPaymentsConfiguration: configurationLoader.configuration) { [weak self] result in
+            switch result {
+            case .success(let eligible):
+                self?.dataSource.isEligibleForCardPresentPayment = eligible
+            case .failure(_):
+                self?.dataSource.isEligibleForCardPresentPayment = false
+            }
+
+            onCompletion()
+        }
+
+        stores.dispatch(action)
+    }
+
+    func loadPaymentGatewayAccounts() {
```

Comment:
I renamed this function to make it more meaningful. Before It was also forcing the refresh of card present payment eligibility when a payment was collected by loading the payment gateway accounts and reloading the data source via the view controller. Now, as `isEligibleForCardPresentPayment` is not a computed property anymore, this wouldn't work.